submissions/dikun/frogger-game/app.js

```diff
@@ -0,0 +1,123 @@
+"use srtict";
+
+const cell = {
+	width: 100,
+	height: 85
+};
+
+const gameField = {
+	top: 0,
+	right: cell.width * 5,
+	bottom: cell.height * 5,
+	left: 0
+};
+
+const playerStart = {
+	x: 202,
+	y: 400
+};
+
+const enemySpeed = {
+	min: 90,
+	max: 250 
+}; 
+
+const sprite = {
+	enemy: 'images/enemy-bug.png',
+	player: 'images/char-boy.png'
+};
+
+const speed = () => enemySpeed.min + Math.floor(Math.random() * enemySpeed.max);
+const crash = 70;
+const border = 20;
+
+const Enemy = function (x, y, speed, player) {
+	this.x = x;
+	this.y = y;
+	this.speed = speed();
+	this.player = player;
+	this.sprite = sprite.enemy;	
+
+	this.checkColission = function () {
+		if(this.player.x + crash > this.x &&
+			this.player.x < this.x + crash &&
+			this.player.y + crash > this.y &&
+			this.player.y < this.y + crash) {
+			this.player.x = playerStart.x;
+			this.player.y = playerStart.y;
+		}
+	};
+
+	this.update = function (dt) {
+		if (this.x > gameField.right) {
+			this.x -= gameField.right;
+			this.speed = speed();
+			}
+		this.x += this.speed * dt;
+		this.checkColission();
+	};
+
+	this.render = function () {ctx.drawImage(Resources.get(this.sprite), this.x, this.y);};
+};
+
+const Player = function () {
+	this.x = playerStart.x;
+	this.y = playerStart.y;
+	this.sprite = sprite.player;	
+
+	this.update = function () {
+		if (this.y  < gameField.top) {			
+		his.x = playerStart.x;
+		this.y = playerStart.y;	
+		alert('You win!');	
+	}};
+
+	this.render = function () {ctx.drawImage(Resources.get(this.sprite), this.x, this.y);};
+
+	this.handleInput = function (keyPress) {
+		if (keyPress == 'left' && this.x > gameField.left + cell.width) {
+			this.x -= cell.width;
+			}
+		if (keyPress == 'right' && this.x < gameField.right - cell.width) {
+			this.x += cell.width;
+		}
+		if (keyPress == 'up' && this.y > gameField.top) {
+			this.y -= cell.height;
+		}
+		if (keyPress == 'down' && this.y < gameField.bottom - cell.height) {
+			this.y += cell.height;
+		}	
+		if (this.y  < gameField.top) {			
+			this.x = playerStart.x;
+			this.y = playerStart.y;	
+			alert('You win!');	
+		}			
+	};
+};
+
+Function.prototype.construct = function (position) {
+	let gameEnteties = Object.create(this.prototype);
+	this.apply(gameEnteties, position);
+	return gameEnteties;
+};
```

Comment:
We do not want to play around with JS built in classes/objects.
If you want a base class, you need a class function like `Player` or `Enemy` and then you bind `Player` and `Enemy` classes with this base class by `Object.create`.